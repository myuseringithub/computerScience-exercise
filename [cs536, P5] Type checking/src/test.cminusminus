// test.cminusminus should contain code with no errors that exercises all of the type-check methods that you wrote for the different AST nodes. This means that it should include (good) examples of every kind of statement and expression.

int x; bool b; void f(){} int g(){return 0;} bool h() {return true;} void m(int x, bool b) {}

void t1() {
    x=x+x-x*x/x;x--;x++;
    b=x>x;b=x<x;b=x<=x;b=x>=x;
    b=x==x;b=x!=x;b=b==b;b=b!=b;
    x=x;b=b;x=g();b=h();
    cin>>x;cin>>b;cout<<x;cout<<b;cout<<""; 
    f();g();h();m(x,b);
    if(b){} if(b){}else{} while(b){} repeat(1){}

    if(b){if(b){while(b){repeat(1){
        x=x+x-x*x/x;x--;x++;
        b=x>x;b=x<x;b=x<=x;b=x>=x;
        b=x==x;b=x!=x;b=b==b;b=b!=b;
        x=x;b=b;x=g();b=h();
        cin>>x;cin>>b;cout<<x;cout<<b;cout<<""; 
        f();g();h();m(x,b);
        if(b){} if(b){}else{} while(b){} repeat(1){}
    }}}else{}} 
}