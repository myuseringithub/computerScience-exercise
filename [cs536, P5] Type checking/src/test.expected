int x;
bool b;
void f() {
}

int g() {
    return 0;
}

bool h() {
    return true;
}

void m(int x, bool b) {
}

void t1() {
    x(int) = ((x(int) + x(int)) - ((x(int) * x(int)) / x(int)));
    x(int)--;
    x(int)++;
    b(bool) = (x(int) > x(int));
    b(bool) = (x(int) < x(int));
    b(bool) = (x(int) <= x(int));
    b(bool) = (x(int) >= x(int));
    b(bool) = (x(int) == x(int));
    b(bool) = (x(int) != x(int));
    b(bool) = (b(bool) == b(bool));
    b(bool) = (b(bool) != b(bool));
    x(int) = x(int);
    b(bool) = b(bool);
    x(int) = g(->int)();
    b(bool) = h(->bool)();
    cin >> x(int);
    cin >> b(bool);
    cout << x(int);
    cout << b(bool);
    cout << "";
    f(->void)();
    g(->int)();
    h(->bool)();
    m(int,bool->void)(x(int), b(bool));
    if (b(bool)) {
    }
    if (b(bool)) {
    }
    else {
    }
    while (b(bool)) {
    }
    repeat (1) {
    }
    if (b(bool)) {
        if (b(bool)) {
            while (b(bool)) {
                repeat (1) {
                    x(int) = ((x(int) + x(int)) - ((x(int) * x(int)) / x(int)));
                    x(int)--;
                    x(int)++;
                    b(bool) = (x(int) > x(int));
                    b(bool) = (x(int) < x(int));
                    b(bool) = (x(int) <= x(int));
                    b(bool) = (x(int) >= x(int));
                    b(bool) = (x(int) == x(int));
                    b(bool) = (x(int) != x(int));
                    b(bool) = (b(bool) == b(bool));
                    b(bool) = (b(bool) != b(bool));
                    x(int) = x(int);
                    b(bool) = b(bool);
                    x(int) = g(->int)();
                    b(bool) = h(->bool)();
                    cin >> x(int);
                    cin >> b(bool);
                    cout << x(int);
                    cout << b(bool);
                    cout << "";
                    f(->void)();
                    g(->int)();
                    h(->bool)();
                    m(int,bool->void)(x(int), b(bool));
                    if (b(bool)) {
                    }
                    if (b(bool)) {
                    }
                    else {
                    }
                    while (b(bool)) {
                    }
                    repeat (1) {
                    }
                }
            }
        }
        else {
        }
    }
}

